catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js

```diff
@@ -14,37 +14,227 @@
+                var newShowValue = this.$el.find('input[type="checkbox"]').is(':checked');
+                this.model.set('show', newShowValue);
+
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                layer.show = newShowValue;
+            },
+            changeUp: function () {
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                this.viewOptions.getCesiumViewer().imageryLayers.raise(layer);
+                this.model.collection.moveUp(this.model);
+            },
+            changeDown: function () {
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                this.viewOptions.getCesiumViewer().imageryLayers.lower(layer);
+                this.model.collection.moveDown(this.model);
+            }
+        });
+
+        PrefsModalView.LayerPickers = Marionette.CollectionView.extend({
+            template: 'layerPickerViewTemplate',
+            childView: PrefsModalView.LayerPicker,
+            tagName: 'table',
+            className: 'table layerPickerTable',
+            childViewOptions: {},
+            initialize: function (options) {
+                // Marionette.CollectionView implicitly handles options.collection
+                this.childViewOptions = options; // pass options to child views.
+            },
+            /*
+             * following code is copy/paste of
+             * https://github.com/marionettejs/backbone.marionette/blob/master/src/collection-view.js#L579-L618
+             * but changed to "reverse" order of rendered list.
+             */
+            attachHtml: function (collectionView, childView, index) {
+                if (collectionView.isBuffering) {
+                    collectionView._bufferedChildren.splice(0, 0, childView); // instead of (index,0,childview)
+                } else {
+                    if (!collectionView.insertAfter(childView, index)) {
+                        collectionView.insertBefore(childView);
+                    }
+                }
             },
+            insertAfter: function(childView, index) {
+                var currentView;
+                var findPosition = this.getOption('sort') && (index < this.children.length - 1);
+                if (findPosition) {
+                    // Find the view after this one
+                    currentView = this.children.find(function(view) {
+                        return view._index === index + 1;
+                    });
+                }
+
+                if (currentView) {
+                    currentView.$el.after(childView.el); // instead of before().
+                    return true;
+                }
+
+                return false;
+            },
+            insertBefore: function(childView) {
+                this.$el.prepend(childView.el); // instead of append().
+            }
+
+        });
+
+        PrefsModalView.LayerTabView = Marionette.LayoutView.extend({
```

Comment:
It would be safer to use OpenLayers in this view. If you go forward with this implementation, you'll also need an OpenLayers one anyways for when WebGL is not supported.